doc/source/handling-dependencies.rst

```diff
@@ -58,6 +58,12 @@ Runtime Environments
 
 A **runtime environment** describes the dependencies your Ray application needs to run, including :ref:`files, packages, environment variables, and more <runtime-environments-api-ref>`.  It is installed dynamically on the cluster at runtime.
 
+.. note::
+
+    This is an emerging feature.
+    For production use, the best practice is to build your files and dependencies into a Docker image and specify the image in your :ref:`Cluster YAML Configuration<cluster-config>`.
```

Comment:
Would runtime env become a long term standard for all dep management in Ray, or would this docker image the best practices even in the future?

Reply:
I think in the long term we will support docker as an option to runtime env, and once that becomes really stable that will be the best practice.  @edoakes does that sound about right?

Reply:
This has different semantics right? IIUC, this approach means you are starting a single container and run all workers there, and runtime env means each of worker will be started on a docker container. Or you meant to support the first use case in the future with runtime env?

Reply:
Yeah good point, it's two different approaches.  I'm not sure about the future plan actually, maybe it's possible to support the first use case with runtime env.  Maybe this is more desirable because it mimics the situation of workers all running on the same machine. 

The current docker support just starts each worker in its own docker container, like you said.

Reply:
Hmm maybe we should consider changing the wording in this case? Maybe we can just say there are several ways to handle dependencies in Ray, and see this link for more details or something like that.